src/main/java/co/cask/hydrator/plugin/batch/DynamoDBBatchSource.java

```diff
@@ -0,0 +1,220 @@
+    private String readThroughput;
+
+    @Description("Read Throughput Percentage for AWS DynamoDB table to connect to. Default is 0.5. (Macro Enabled)")
+    @Nullable
+    @Macro
+    private String readThroughputPercentage;
+
+    public DynamoDBConfig(String referenceName, String accessKey, String secretAccessKey, @Nullable String regionId,
+                          @Nullable String endpointUrl, String tableName, String query, @Nullable String filterQuery,
+                          @Nullable String nameMappings, String valueMappings, String placeholderType,
+                          @Nullable String readThroughput, @Nullable String readThroughputPercentage) {
+      super(referenceName);
+      this.accessKey = accessKey;
+      this.secretAccessKey = secretAccessKey;
+      this.regionId = regionId;
+      this.endpointUrl = endpointUrl;
+      this.tableName = tableName;
+      this.query = query;
+      this.filterQuery = filterQuery;
+      this.nameMappings = nameMappings;
+      this.valueMappings = valueMappings;
+      this.placeholderType = placeholderType;
+      this.readThroughput = readThroughput;
+      this.readThroughputPercentage = readThroughputPercentage;
+    }
+
+    /**
+     * Validates whether the table name follows the DynamoDB naming rules and conventions or not.
+     */
+    private void validateTableName() {
+      int tableNameLength = tableName.length();
+      if (tableNameLength < 3 || tableNameLength > 255) {
+        throw new IllegalArgumentException(
+          String.format("Table name '%s' does not follow the DynamoDB naming rules. Table name must be between 3 and " +
+                          "255 characters long.", tableName));
+
+      }
+
+      String pattern = "^[a-zA-Z0-9_.-]+$";
+      Pattern patternObj = Pattern.compile(pattern);
+
+      if (!Strings.isNullOrEmpty(tableName)) {
+        Matcher matcher = patternObj.matcher(tableName);
+        if (!matcher.find()) {
+          throw new IllegalArgumentException(
+            String.format("Table name '%s' does not follow the DynamoDB naming rules. Table names can contain only " +
+                            "the following characters: 'a-z, A-Z, 0-9, underscore(_), dot(.) and dash(-)'.",
+                          tableName));
+        }
+      }
+    }
+  }
+
+  /**
+   * Input format provider for DynamoDBBatchSource.
+   */
+  private static class DynamoDBInputFormatProvider implements InputFormatProvider {
+    private Map<String, String> conf;
+
+    DynamoDBInputFormatProvider(DynamoDBConfig dynamoDBConfig) {
+      this.conf = new HashMap<>();
+      conf.put(DynamoDBConstants.ACCESS_KEY, dynamoDBConfig.accessKey);
+      conf.put(DynamoDBConstants.SECRET_ACCESS_KEY, dynamoDBConfig.secretAccessKey);
+      conf.put(DynamoDBConstants.REGION_ID, dynamoDBConfig.regionId == null ? "" : dynamoDBConfig.regionId);
+      conf.put(DynamoDBConstants.ENDPOINT_URL, dynamoDBConfig.endpointUrl == null ? "" : dynamoDBConfig.endpointUrl);
```

Comment:
Yes, it is possible.
User will always have to provide access and secret key and even if the user does not provide any region, default region is 'us-west-2'.
So, if user does not provide regionID and endpointUrl, then using accesKey, secretAccessKey and default region, the DynamoDB client will get created, considering that user expects the region to be us-west-2.
Also, if user does not provide the endpointUrl, the program will reconstructs the Url as per the keys and region.

Updated the description for endpointUrl accordingly.